pkg/strategy/drift/strategy.go

```diff
@@ -572,35 +560,39 @@ func (s *Strategy) klineHandler1m(ctx context.Context, kline types.KLine) {
 	if s.Status != types.StrategyStatusRunning {
 		return
 	}
-	if s.NoTrailingStopLoss || s.TrailingStopLossType == "realtime" {
-		return
-	}
 	// for doing the trailing stoploss during backtesting
 	atr := s.atr.Last()
 	price := s.getLastPrice()
 	pricef := price.Float64()
 	stoploss := s.StopLoss.Float64()
 
-	var err error
+	lowf := math.Min(kline.Low.Float64(), pricef)
+	highf := math.Max(kline.High.Float64(), pricef)
+	s.positionLock.Lock()
+	if s.lowestPrice > 0 && lowf < s.lowestPrice {
+		s.lowestPrice = lowf
+	}
+	if s.highestPrice > 0 && highf > s.highestPrice {
+		s.highestPrice = highf
+	}
+
 	numPending := 0
+	var err error
 	if numPending, err = s.smartCancel(ctx, pricef, atr); err != nil {
 		log.WithError(err).Errorf("cannot cancel orders")
+		s.positionLock.Unlock()
 		return
 	}
 	if numPending > 0 {
-		log.Infof("pending orders: %d, exit", numPending)
+		s.positionLock.Unlock()
```

Comment:
types.Position has it's lock, maybe you can reuse it?

Reply:
I believe it's for different purpose.  
this position lock is mainly for price updates (buyPrice, sellPrice, lowestPrice, highestPrice), so should be fine.